common/lsp/lsp-file-utils_test.cc

```diff
@@ -0,0 +1,83 @@
+// Copyright 2023 The Verible Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#include "common/lsp/lsp-file-utils.h"
+
+#include "gtest/gtest.h"
+
+TEST(URIDecodingTest, SimplePathDecoding) {
+  std::string exemplar = "/home/user/test-project/add.sv";
+  std::string result =
+      verible::lsp::LSPUriToPath("file:///home/user/test-project/add.sv");
+
+  ASSERT_EQ(exemplar, result);
+}
+
+TEST(URIDecodingTest, WindowsPathDecoding) {
```

Comment:
Instead of first having all the decoding tests, then the encoding tests, I'd probably group the tests so that encoding and decoding are close together; that makes it easier to look at them in one go.

WindowsPathDecoding
WindowsPathEncoding

PathWithSpaceDecoding
PathWithSpacesEncoding
...